src/goto-instrument/code_contracts.cpp

```diff
@@ -984,14 +984,14 @@ void code_contractst::add_contract_check(
   PRECONDITION(!dest.instructions.empty());
 
   const symbolt &function_symbol = ns.lookup(mangled_fun);
-  auto code_type = to_code_with_contract_type(function_symbol.type);
+  const auto &code_type = to_code_with_contract_type(function_symbol.type);
 
-  exprt &assigns = code_type.assigns();
-  exprt &requires = code_type.requires();
-  exprt &ensures = code_type.ensures();
+  exprt assigns = code_type.assigns();
+  exprt requires = conjunction(code_type.requires());
+  exprt ensures = conjunction(code_type.ensures());
```

Comment:
I wonder if it would be better to create separate assertions instead of just creating a single conjunctive assertion.

I'm thinking about failure reporting. Would we be able to report the right source location if say the 3rd ensures clause failed? Since we would generate a single assertion, we would only generate one source location for the failure, and I think that would be the location of the first ensures clause?

Reply:
Yes, please do PR!

Reply:
Okay let's merge this since it fixes the issue of contracts being silently ignored. I / Felipe would then make a separate PR to improve error reporting.